src/protocols/BaseProtocolAdapter.ts

```diff
@@ -65,7 +66,7 @@ export abstract class BaseProtocolAdapter {
    * @param seed 12 word seed array buffer
    * @returns total number of used accounts
    */
-  abstract discoverAccounts(seed: Uint8Array): Promise<number>;
+  abstract discoverAccounts(seed: Uint8Array): Promise<DiscoveredAccounts>;
```

Comment:
I think it makes no sense to return the protocol ID by the `discoverAccounts` method.